src/server/stream_family.cc

```diff
@@ -1797,19 +1797,18 @@ OpResult<PendingResult> OpPending(const OpArgs& op_args, string_view key, const
   return result;
 }
 
-void CreateGroup(CmdArgList args, string_view key, ConnectionContext* cntx) {
-  if (args.size() < 2)
-    return cntx->SendError(UnknownSubCmd("CREATE", "XGROUP"));
+void CreateGroup(facade::CmdArgParser* parser, ConnectionContext* cntx) {
+  auto key = parser->Next();
 
   CreateOpts opts;
-  opts.gname = ArgS(args, 0);
-  opts.id = ArgS(args, 1);
-  if (args.size() >= 3) {
-    ToUpper(&args[2]);
-    if (ArgS(args, 2) == "MKSTREAM")
-      opts.flags |= kCreateOptMkstream;
+  std::tie(opts.gname, opts.id) = parser->Next<string_view, string_view>();
+  if (parser->Check("MKSTREAM")) {
+    opts.flags |= kCreateOptMkstream;
```

Comment:
opts.flags |= bool(check()) if you want to save two lines 🤓🤓

Reply:
kCreateOptMkstream can be not 1, so it's dangerous
in this case the correct one is opts.flags |= bool(check()) * kCreateOptMkstream